ext/openssl/openssl.c

```diff
@@ -1018,9 +1099,47 @@ PHP_MINIT_FUNCTION(openssl)
 {
 	char * config_filename;
 
-	le_key = zend_register_list_destructors_ex(php_openssl_pkey_free, NULL, "OpenSSL key", module_number);
-	le_x509 = zend_register_list_destructors_ex(php_openssl_x509_free, NULL, "OpenSSL X.509", module_number);
-	le_csr = zend_register_list_destructors_ex(php_openssl_csr_free, NULL, "OpenSSL X.509 CSR", module_number);
+	zend_class_entry ce;
+	INIT_CLASS_ENTRY(ce, "X509Certificate", class_X509Certificate_methods);
```

Comment:
Can you please prefix it with OpenSSL. We could even consider namespace here. Altough not sure if it's not too controversial as I don't think we have another core extension using namespaces. But think it would make sense. If you don't want to use namespace, please at least prefix the name.

Reply:
We do have `FFI` which uses namespaces :)

But generally I agree, putting these into namespaces would be a bit exceptional (although I'd love having an `OpenSSL` namespace, without any `Ext\` or `PHP\` prefix). Unfortunately, I'm afraid we shouldn't do this until we can find out the details _as a project_. @nikic do you have any suggestion here?

Reply:
Yeah, as far as this PR is concerned, we can't use a namespace. Thanks for the FFI example though, I forgot that we actually have precedent for unvendored namespaces in php-src.

Reply:
Ok in that case please prefix the name with `OpenSSL`.